main.cpp

```diff
@@ -65,6 +65,51 @@ void set_console_state(console_state new_st)
     }
 }
 
+static constexpr uint32_t LLAMA_TRACE_VERSION = 0;
+static constexpr uint32_t LLAMA_TRACE_MAGIC = 0x67676d74; // 'ggmt' in hex
```

Comment:
I would put this constant (and possibly the version) into `llama.h`.

Reply:
The format is not really meant to be part of the library. 
However, it could live under `common.h` (was `utils.h`).
Exposing the `trace_*` function as a C API would require switching to stdio and `FILE*` handles.